File: Leetcode_重刷紀錄/0001.Problems1768. Merge Strings Alternately/MergeStringsAlternately.java
public class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder result = new StringBuilder();
        int i = 0, len1 = word1.length(), len2 = word2.length();
        int maxLen = Math.max(len1, len2);  // 新宣告的參數
        
        while (i < maxLen) {
            if (i < len1) {
                result.append(word1.charAt(i));
            }
            if (i < len2) {
                result.append(word2.charAt(i));
            }
            i++;
        }
        
        return result.toString();
    }
}

File: Leetcode_重刷紀錄/0001.Problems1768. Merge Strings Alternately/MergeStringsAlternately.cpp
lass Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string result;
        int len1 = word1.size(), len2 = word2.size(),resultlen=len1+len2;
        result.reserve(len1 + len2);  // 預分配記憶體
        
        int i = 0, j = 0;
        while (i + j< resultlen) {
            if (i < len1) {
                result.push_back(word1[i++]);
            }
            if (j < len2) {
                result.push_back(word2[j++]);
            }
        }
        return result;
    }
};